src/checkers/inference/model/LubVariableSlot.java

```diff
@@ -0,0 +1,76 @@
+package checkers.inference.model;
+
+/**
+ * LubVariableSlot models the least-upper-bounds of two other slots.
+ */
+public class LubVariableSlot extends VariableSlot {
+
+    private final Slot left;
+    private final Slot right;
+
+    public LubVariableSlot(AnnotationLocation location, int id, Slot left, Slot right) {
+        super(location, id);
+        this.left = left;
+        this.right = right;
+    }
+
+    @Override
+    public Kind getKind() {
+        return Kind.LUB_VARIABLE;
+    }
+
+    @Override
+    public <S, T> S serialize(Serializer<S, T> serializer) {
+        return serializer.serialize(this);
+    }
+
+    public Slot getLeft() {
+        return left;
+    }
+
+    public Slot getRight() {
+        return right;
+    }
+
+    /**
+     * LubVariableSlot should never be inserted into the source code. record
+     * does not correspond to an annotatable position.
+     *
+     * @return false
+     */
+    @Override
+    public boolean isInsertable() {
+        return false;
+    }
+
+    @Override
+    public int hashCode() {
+        final int prime = 31;
```

Comment:
Use hashcodeutils

Reply:
This implementation is consistent with other subclasses of `VariableSlot`. Keeping the same style is also important.

Reply:
Do this later.